trigger.js

```diff
@@ -206,6 +215,44 @@ class TriggerHappy {
         Hooks.once('updateToken', () => this._executeTriggers(triggers));
         return true;
     }
+  
+    _onPreUpdateToken(scene, id, update) {
+        if (!scene.isView) return true;
+        if (update.x === undefined && update.y === undefined) return true;
+        const token = scene.data.tokens.find(t => t._id === update._id);
+        if (token.hidden) return true; // don't stop ivnisible tokens?
+
+        // Get all trigger tokens in scene
+        let targets = this._getTriggerTokens(canvas.tokens.placeables, this.triggers, 'capture');
+        if (!targets) return true;
+
+        let x = update.x || token.x;
+        let  y = update.y || token.y;
+        let tw = token.width * canvas.scene.data.grid / 2;
+        let th = token.height * canvas.scene.data.grid / 2;
+        let motion = new Ray({x: token.x + tw, y: token.y + th}, {x: x + tw, y: y + th});
+
+        // don't trigger on tokens that are already captured
+        targets = targets.filter(target => token.x !== target.x + target.w  / 2 - tw || token.y !== target.y + target.h / 2 - th)
+        
+        // sort list by distance from start token position
+        targets.sort((a , b) => dist(token.x, token.y, b.x, b.y) - dist(token.x, token.y, a.x, a.y));
+        
+        targets.forEach(target => {
+            // test motion vs token diagonals
+            if (motion.intersectSegment([target.x, target.y, target.x + target.w, target.y + target.h])
+            || motion.intersectSegment([target.x, target.y + target.h, target.x + target.w, target.y])) {
+                update.x = target.x + target.w / 2 - tw;
+                update.y = target.y + target.h / 2 - th;

```

Comment:
Use `target.center`

Reply:
yeah. This all started when I was getting token.data instead of actual tokens and none of center, w or anything else was available. For some reason I did not go back and revisit everywhere once it was working with actual tokens.

Reply:
Humm.. hadn't realized `token` is from `scene.data.tokens` instead of `canvas.tokens`... I thought I added the `scene.isView` check explicitly so I could use `canvas.tokens` and know it's not the wrong scene... I don't know why that's not it, I'll have to revisit and clean up the code later.